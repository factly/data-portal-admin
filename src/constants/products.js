export const LOADING_PRODUCTS = 'LOADING_PRODUCTS';
export const LOAD_PRODUCTS_SUCCESS = 'LOAD_PRODUCTS_SUCCESS';
export const ADD_PRODUCTS_LIST_REQUEST = 'ADD_PRODUCTS_LIST_REQUEST';
export const SET_PRODUCTS_LIST_CURRENT_PAGE = 'SET_PRODUCTS_LIST_CURRENT_PAGE';
export const SET_PRODUCTS_LIST_TOTAL = 'SET_PRODUCTS_LIST_TOTAL';
export const LOAD_PRODUCTS_FAILURE = 'LOAD_PRODUCTS_FAILURE';
export const CREATING_PRODUCT = 'CREATING_PRODUCT';
export const CREATE_PRODUCT_SUCCESS = 'CREATE_PRODUCT_SUCCESS';
export const CREATE_PRODUCT_FAILURE = 'CREATE_PRODUCT_FAILURE';
export const LOADING_PRODUCT_DETAILS = 'LOADING_PRODUCT_DETAILS';
export const GET_PRODUCT_DETAILS_SUCCESS = 'GET_PRODUCT_DETAILS_SUCCESS';
export const GET_PRODUCT_DETAILS_FAILURE = 'GET_PRODUCT_DETAILS_FAILURE';
export const UPDATING_PRODUCT = 'UPDATING_PRODUCT';
export const UPDATE_PRODUCT_SUCCESS = 'UPDATE_PRODUCT_SUCCESS';
export const UPDATE_PRODUCT_FAILURE = 'UPDATE_PRODUCT_FAILURE';
export const DELETING_PRODUCT = 'DELETING_PRODUCT';
export const DELETE_PRODUCT_SUCCESS = 'DELETE_PRODUCT_SUCCESS';
export const DELETE_PRODUCT_FAILURE = 'DELETE_PRODUCT_FAILURE';

export const baseUrl = `${process.env.REACT_APP_API_URL}/products`;
