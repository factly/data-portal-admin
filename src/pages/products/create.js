import React, { useState } from 'react';
import { Form, Input, Button, notification, Select } from 'antd';

const { Option } = Select;

const formItemLayout = {
  labelCol: { span: 6 },
  wrapperCol: { span: 14 },
};

const ProductCreate = (props) => {
  const [tags, setTags] = useState([]);
  const [categories, setCategories] = useState([]);
  const [currencies, setCurrencies] = useState([]);
  const [productType, setProductType] = useState([]);

  React.useEffect(() => {
    fetch(process.env.REACT_APP_API_URL + '/tags')
      .then((data) => data.json())
      .then((data) => {
        setTags(data.nodes);
      });
    fetch(process.env.REACT_APP_API_URL + '/categories')
      .then((data) => data.json())
      .then((data) => {
        setCategories(data.nodes);
      });
    fetch(process.env.REACT_APP_API_URL + '/currencies')
      .then((data) => data.json())
      .then((data) => {
        setCurrencies(data.nodes);
      });
    fetch(process.env.REACT_APP_API_URL + '/types')
      .then((data) => data.json())
      .then((data) => {
        setProductType(data.nodes);
      });
  }, []);

  function handleChange(value) {
    console.log(`selected ${value}`);
  }

  const onFinish = (values) => {
    values.currency_id = parseInt(values.currency_id);
    values.product_type_id = parseInt(values.product_type_id);
    values.price = parseInt(values.price);
    values.category_ids = values.category_ids.map((c) => parseInt(c));
    values.tag_ids = values.tag_ids.map((t) => parseInt(t));
    fetch(process.env.REACT_APP_API_URL + '/products', {
      method: 'POST',
      body: JSON.stringify(values),
    })
      .then((res) => {
        if (res.status === 201) {
          return res.json();
        } else {
          throw new Error(res.status);
        }
      })
      .then((_) => {
        notification.success({
          message: 'Success',
          description: 'Product succesfully added',
        });
        props.history.push('/products');
      })
      .catch((res) => {
        notification.error({
          message: 'Error',
          description: 'Something went wrong',
        });
      });
  };

  return (
    <Form name="products_create" {...formItemLayout} onFinish={onFinish}>
      <Form.Item
        label="Title"
        name="title"
        rules={[
          {
            required: true,
            message: 'Please enter title!',
          },
        ]}
      >
        <Input placeholder="Ex. Crime In India" />
      </Form.Item>

      <Form.Item
        label="Slug"
        name="slug"
        rules={[
          {
            required: true,
            message: 'Please enter slug!',
          },
        ]}
      >
        <Input placeholder="Ex. crime-in-india" />
      </Form.Item>

      <Form.Item
        label="Currency"
        name="currency_id"
        rules={[
          {
            required: true,
            message: 'Please enter currency!',
          },
        ]}
      >
        <Select
          showSearch
          style={{ width: '100%' }}
          placeholder="Select currency"
          onChange={handleChange}
          filterOption={(input, option) =>
            option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
          }
        >
          {currencies.length > 0
            ? currencies.map((c) => <Option key={c.id}>{c.iso_code}</Option>)
            : []}
        </Select>
      </Form.Item>

      <Form.Item
        label="Price"
        name="price"
        rules={[
          {
            required: true,
            message: 'Please enter price!',
          },
        ]}
      >
        <Input placeholder="Ex. 1999" />
      </Form.Item>

      <Form.Item
        label="Category"
        name="category_ids"
        rules={[
          {
            required: true,
            message: 'Please select atleast one category!',
          },
        ]}
      >
        <Select
          mode="tags"
          style={{ width: '100%' }}
          placeholder="Select categories"
          onChange={handleChange}
          filterOption={(input, option) =>
            option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
          }
        >
          {categories.length > 0 ? categories.map((c) => <Option key={c.id}>{c.slug}</Option>) : []}
        </Select>
      </Form.Item>

      <Form.Item
        label="Tag"
        name="tag_ids"
        rules={[
          {
            required: true,
            message: 'Please select atleast one tag!',
          },
        ]}
      >
        <Select
          mode="tags"
          style={{ width: '100%' }}
          placeholder="Select tags"
          onChange={handleChange}
          filterOption={(input, option) =>
            option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
          }
        >
          {tags.length > 0 ? tags.map((t) => <Option key={t.id}>{t.slug}</Option>) : []}
        </Select>
      </Form.Item>

      <Form.Item
        label="Status"
        name="status"
        rules={[
          {
            required: true,
            message: 'Please enter status!',
          },
        ]}
      >
        <Select
          showSearch
          style={{ width: '100%' }}
          placeholder="Select status"
          defaultValue={'Hide'}
          onChange={handleChange}
          filterOption={(input, option) =>
            option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
          }
        >
          <Option key={'Hide'}>Hide</Option>
          <Option key={'Unhide'}>Unhide</Option>
        </Select>
      </Form.Item>

      <Form.Item
        label="Type"
        name="product_type_id"
        rules={[
          {
            required: true,
            message: 'Please enter type!',
          },
        ]}
      >
        <Select
          showSearch
          style={{ width: '100%' }}
          placeholder="Select type"
          onChange={handleChange}
          filterOption={(input, option) =>
            option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
          }
        >
          {productType.length > 0
            ? productType.map((p) => <Option key={p.id}>{p.name}</Option>)
            : []}
        </Select>
      </Form.Item>

      <Form.Item wrapperCol={{ span: 12, offset: 6 }}>
        <Button type="primary" htmlType="submit">
          Submit
        </Button>
      </Form.Item>
    </Form>
  );
};

export default ProductCreate;
