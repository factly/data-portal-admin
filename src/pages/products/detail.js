import React from 'react';
import { useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';

import ProductDetails from './components/ProductDetails';

const ProductDetail = () => {
  const { id } = useParams();
  const { product, currencies, tags } = useSelector(({ products, currencies, tags }) => ({
    currencies: currencies.items,
    tags: tags.items,
    product: products.items[id],
  }));

  const currencyCode = currencies[product.currency_id].iso_code;
  const productTags = [];
  product.tags.forEach((id) => productTags.push(tags[id]));

  return <ProductDetails product={product} productTags={productTags} currencyCode={currencyCode} />;
};

export default ProductDetail;
