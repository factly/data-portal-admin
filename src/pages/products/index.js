import React, { useState } from 'react';
import { Table, Input, Button, Popconfirm, Form, notification } from 'antd';
import moment from 'moment';
import { Link } from 'react-router-dom';
import { EditOutlined, DeleteOutlined } from '@ant-design/icons';

const EditableCell = ({ editing, dataIndex, title, record, index, children, ...restProps }) => {
  return (
    <td {...restProps}>
      {editing ? (
        <Form.Item
          name={dataIndex}
          style={{
            margin: 0,
          }}
          rules={[
            {
              required: true,
              message: `Please enter ${title}!`,
            },
          ]}
        >
          <Input />
        </Form.Item>
      ) : (
        children
      )}
    </td>
  );
};

const Products = (props) => {
  const [form] = Form.useForm();
  const [data, setData] = useState([]);
  const [total, setTotal] = useState(0);
  const [editingKey, setEditingKey] = useState('');

  React.useEffect(() => {
    fetch(process.env.REACT_APP_API_URL + '/products')
      .then((data) => data.json())
      .then((data) => {
        setData(data.nodes);
        setTotal(data.total);
      });
  }, []);

  const get = (page, limit) => {
    cancel();
    fetch(process.env.REACT_APP_API_URL + '/products?page=' + page + '&limit=' + limit)
      .then((data) => data.json())
      .then((data) => {
        setData(data.nodes);
        setTotal(data.total);
      });
  };

  const isEditing = (record) => record.id === editingKey;

  const cancel = () => {
    setEditingKey('');
  };

  const deleteProduct = (key) => {
    const index = data.findIndex((item) => item.id === key);
    if (index > -1) {
      fetch(process.env.REACT_APP_API_URL + '/products/' + key, {
        method: 'DELETE',
      })
        .then((res) => {
          if (res.status === 200) {
            const newData = [...data];
            newData.splice(index, 1);
            setData(newData);
            notification.success({
              message: 'Success',
              description: 'Product succesfully deleted',
            });
          }
        })
        .catch((err) => {
          notification.error({
            message: 'Error',
            description: 'Something went wrong',
          });
        });
    }
  };

  const columns = [
    {
      title: 'Title',
      dataIndex: 'title',
      width: '10%',
    },
    {
      title: 'Slug',
      dataIndex: 'slug',
      width: '10%',
    },
    {
      title: 'Currency',
      render: (record) => record.currency.iso_code,
      width: '10%',
    },
    {
      title: 'Price',
      dataIndex: 'price',
      width: '10%',
    },
    {
      title: 'Status',
      render: (record) => record.status,
      width: '10%',
    },
    {
      title: 'Categories',
      render: (record) => record.categories.map((c) => c.title).join(', '),
      width: '20%',
    },
    {
      title: 'Tags',
      render: (record) => record.tags.map((t) => t.title).join(', '),
      width: '20%',
    },
    {
      title: 'Created At',
      dataIndex: 'CreatedAt',
      width: '10%',
      render: (_, record) => {
        return <span title={record.CreatedAt}>{moment(record.CreatedAt).fromNow()}</span>;
      },
    },
    {
      title: 'Operation',
      dataIndex: 'operation',
      render: (_, record) => {
        return (
          <span>
            <Button
              type="primary"
              icon={<EditOutlined />}
              disabled={editingKey !== ''}
              onClick={() => {
                props.history.push(`/products/detail/${record.id}`);
              }}
              style={{
                marginRight: 8,
              }}
            >
              Edit
            </Button>
            <Popconfirm
              disabled={editingKey !== ''}
              title="Sure to delete?"
              onConfirm={() => deleteProduct(record.id)}
            >
              <Button disabled={editingKey !== ''} icon={<DeleteOutlined />}>
                Delete
              </Button>
            </Popconfirm>
          </span>
        );
      },
    },
  ];

  const mergedColumns = columns.map((col) => {
    if (!col.editable) {
      return col;
    }

    return {
      ...col,
      onCell: (record) => ({
        record,
        dataIndex: col.dataIndex,
        title: col.title,
        editing: isEditing(record),
      }),
    };
  });

  return (
    <div>
      <Link to={'/products/create'}>
        <Button type="primary" style={{ marginBottom: 16 }}>
          Add Product
        </Button>
      </Link>
      <Form form={form} component={false}>
        <Table
          components={{
            body: {
              cell: EditableCell,
            },
          }}
          bordered
          rowKey="id"
          dataSource={data}
          columns={mergedColumns}
          rowClassName="editable-row"
          pagination={{
            defaultPageSize: 5,
            onChange: get,
            total: total,
          }}
        />
      </Form>
    </div>
  );
};

export default Products;
