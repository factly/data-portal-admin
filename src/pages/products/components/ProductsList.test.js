import React from 'react';
import { BrowserRouter as Router, Link, useHistory } from 'react-router-dom';
import renderer, { act } from 'react-test-renderer';
import { useSelector, useDispatch } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow, mount } from 'enzyme';
import { Popconfirm, Button, List, Card } from 'antd';

import '../../../matchMedia.mock';
import ProductList from './ProductsList';
import { loadProducts, deleteProduct } from '../../../actions/products';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn(),
}));
jest.mock('../../../actions/products', () => ({
  loadProducts: jest.fn(),
  deleteProduct: jest.fn(),
}));

describe('Products List component', () => {
  let store;
  let mockedDispatch;
  const product = {
    id: 1,
    title: 'title',
    slug: 'slug',
    price: 100,
    status: 'status',
    tags: [1],
    currency_id: 1,
    created_at: '2020-12-12',
  };

  describe('snapshot testing', () => {
    beforeEach(() => {
      store = mockStore({});
      store.dispatch = jest.fn();
      mockedDispatch = jest.fn();
      useDispatch.mockReturnValue(mockedDispatch);
    });
    it('should render the component', () => {
      useSelector.mockImplementation((state) => ({}));
      const tree = renderer.create(<ProductList />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(useSelector).toHaveBeenCalled();
    });
    it('should match component when loading', () => {
      useSelector.mockImplementation((state) => ({
        data: [],
        currencies: {},
        tags: {},
        total: 0,
      }));
      const tree = renderer.create(<ProductList />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(useSelector).toHaveBeenCalled();
    });
    it('should match component with products', () => {
      useSelector.mockImplementation((state) => ({
        data: [product],
        total: 1,
        currencies: { 1: { id: 1, iso_code: 'INR' } },
        tags: { 1: { id: 1, title: 'tag1' } },
      }));

      let component;
      act(() => {
        component = renderer.create(
          <Router>
            <ProductList />
          </Router>,
        );
      });
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();

      expect(useSelector).toHaveBeenCalled();
      expect(mockedDispatch).toHaveBeenCalledTimes(1);
      expect(useSelector).toHaveReturnedWith({
        data: [product],
        total: 1,
        currencies: { 1: { id: 1, iso_code: 'INR' } },
        tags: { 1: { id: 1, title: 'tag1' } },
      });
      expect(loadProducts).toHaveBeenCalledWith(1, 5);
    });
  });
  describe('component testing', () => {
    beforeEach(() => {
      jest.clearAllMocks();
      mockedDispatch = jest.fn(() => new Promise((resolve) => resolve(true)));
      useDispatch.mockReturnValue(mockedDispatch);
    });
    it('should change the page', () => {
      useSelector.mockImplementation((state) => ({}));

      const wrapper = shallow(<ProductList />);
      const list = wrapper.find(List);
      list.props().pagination.onChange(2);
      wrapper.update();
      const updatedList = wrapper.find(List);
      expect(updatedList.props().pagination.current).toEqual(2);
    });
    it('should edit product', () => {
      const push = jest.fn();
      useHistory.mockReturnValueOnce({ push });

      useSelector.mockImplementation((state) => ({
        data: [product],
        total: 1,
        currencies: { 1: { id: 1, iso_code: 'INR' } },
        tags: { 1: { id: 1, title: 'tag1' } },
      }));

      const wrapper = mount(
        <Router>
          <ProductList />
        </Router>,
      );
      const card = wrapper.find(Card).at(0);
      const button = card.find('li').at(0).find('span').at(1);
      button.simulate('click');
      expect(push).toHaveBeenCalledWith('/products/1/edit');
    });
    it('should delete product', () => {
      useSelector.mockImplementation((state) => ({
        data: [product],
        total: 1,
        currencies: { 1: { id: 1, iso_code: 'INR' } },
        tags: { 1: { id: 1, title: 'tag1' } },
      }));

      const wrapper = mount(
        <Router>
          <ProductList />
        </Router>,
      );
      const card = wrapper.find(Card).at(0);
      const button = card.find('li').at(1).find('span').at(1);

      button.simulate('click');
      const popconfirm = wrapper.find(Popconfirm);
      popconfirm
        .findWhere((item) => item.type() === 'button' && item.text() === 'OK')
        .simulate('click');
      expect(deleteProduct).toHaveBeenCalled();
      expect(deleteProduct).toHaveBeenCalledWith(1);
      expect(loadProducts).toHaveBeenCalledWith(1, 5);
    });
    it('should have no delete and edit buttons', () => {
      useSelector.mockImplementation((state) => ({}));

      const wrapper = mount(
        <Router>
          <ProductList />
        </Router>,
      );

      const button = wrapper.find(Button);
      expect(button.length).toEqual(0);
    });
  });
});
