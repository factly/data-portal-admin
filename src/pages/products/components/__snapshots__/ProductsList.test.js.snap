// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products List component snapshot testing should match component when loading 1`] = `
<div
  className="ant-list ant-list-split ant-list-something-after-last-item"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-list-empty-text"
      >
        <div
          className="ant-empty ant-empty-normal"
        >
          <div
            className="ant-empty-image"
          >
            <svg
              className="ant-empty-img-simple"
              height="41"
              viewBox="0 0 64 41"
              width="64"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fillRule="evenodd"
                transform="translate(0 1)"
              >
                <ellipse
                  className="ant-empty-img-simple-ellipse"
                  cx="32"
                  cy="33"
                  rx="32"
                  ry="7"
                />
                <g
                  className="ant-empty-img-simple-g"
                  fillRule="nonzero"
                >
                  <path
                    d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                  />
                  <path
                    className="ant-empty-img-simple-path"
                    d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                  />
                </g>
              </g>
            </svg>
          </div>
          <p
            className="ant-empty-description"
          >
            No Data
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-list-pagination"
  >
    <ul
      className="ant-pagination"
      style={Object {}}
      unselectable="unselectable"
    >
      <li
        aria-disabled={true}
        className="ant-pagination-prev ant-pagination-disabled"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={null}
        title="Previous Page"
      >
        <a
          className="ant-pagination-item-link"
          disabled={true}
        >
          <span
            aria-label="left"
            className="anticon anticon-left"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="left"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
              />
            </svg>
          </span>
        </a>
      </li>
      <li
        className="ant-pagination-item ant-pagination-item-0 ant-pagination-disabled ant-pagination-item-disabled"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
        title={0}
      >
        <a>
          0
        </a>
      </li>
      <li
        aria-disabled={true}
        className="ant-pagination-next ant-pagination-disabled"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={null}
        title="Next Page"
      >
        <a
          className="ant-pagination-item-link"
          disabled={true}
        >
          <span
            aria-label="right"
            className="anticon anticon-right"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
              />
            </svg>
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Products List component snapshot testing should match component with products 1`] = `
<div
  className="ant-list ant-list-split ant-list-something-after-last-item"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <ul
        className="ant-list-items"
      >
        <li
          className="ant-list-item"
        >
          <div
            className="ant-card ant-card-bordered"
          >
            <div
              className="ant-card-body"
              style={Object {}}
            >
              <div
                className="ant-descriptions"
              >
                <div
                  className="ant-descriptions-title"
                >
                  <a
                    href="/products/1"
                    onClick={[Function]}
                  >
                    title
                  </a>
                </div>
                <div
                  className="ant-descriptions-view"
                >
                  <table>
                    <tbody>
                      <tr
                        className="ant-descriptions-row"
                      >
                        <td
                          className="ant-descriptions-item"
                          colSpan={1}
                        >
                          <span
                            className="ant-descriptions-item-label ant-descriptions-item-colon"
                          >
                            Slug
                          </span>
                          <span
                            className="ant-descriptions-item-content"
                          >
                            slug
                          </span>
                        </td>
                        <td
                          className="ant-descriptions-item"
                          colSpan={1}
                        >
                          <span
                            className="ant-descriptions-item-label ant-descriptions-item-colon"
                          >
                            Price
                          </span>
                          <span
                            className="ant-descriptions-item-content"
                          >
                            100
                             
                            INR
                          </span>
                        </td>
                        <td
                          className="ant-descriptions-item"
                          colSpan={1}
                        >
                          <span
                            className="ant-descriptions-item-label ant-descriptions-item-colon"
                          >
                            Status
                          </span>
                          <span
                            className="ant-descriptions-item-content"
                          >
                            status
                          </span>
                        </td>
                      </tr>
                      <tr
                        className="ant-descriptions-row"
                      >
                        <td
                          className="ant-descriptions-item"
                          colSpan={1}
                        >
                          <span
                            className="ant-descriptions-item-label ant-descriptions-item-colon"
                          >
                            Tags
                          </span>
                          <span
                            className="ant-descriptions-item-content"
                          >
                            <span
                              className="ant-tag"
                              style={
                                Object {
                                  "backgroundColor": undefined,
                                }
                              }
                            >
                              tag1
                            </span>
                          </span>
                        </td>
                        <td
                          className="ant-descriptions-item"
                          colSpan={2}
                        >
                          <span
                            className="ant-descriptions-item-label ant-descriptions-item-colon"
                          >
                            Created At
                          </span>
                          <span
                            className="ant-descriptions-item-content"
                          >
                            in 4 months
                          </span>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
            <ul
              className="ant-card-actions"
            >
              <li
                style={
                  Object {
                    "width": "50%",
                  }
                }
              >
                <span>
                  <span
                    aria-label="edit"
                    className="anticon anticon-edit"
                    onClick={[Function]}
                    role="img"
                    tabIndex={-1}
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="edit"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                      />
                    </svg>
                  </span>
                </span>
              </li>
              <li
                style={
                  Object {
                    "width": "50%",
                  }
                }
              >
                <span>
                  <span
                    aria-label="delete"
                    className="anticon anticon-delete"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onTouchStart={[Function]}
                    role="img"
                    tabIndex={-1}
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="delete"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                      />
                    </svg>
                  </span>
                </span>
              </li>
            </ul>
          </div>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="ant-list-pagination"
  >
    <ul
      className="ant-pagination"
      style={Object {}}
      unselectable="unselectable"
    >
      <li
        aria-disabled={true}
        className="ant-pagination-prev ant-pagination-disabled"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={null}
        title="Previous Page"
      >
        <a
          className="ant-pagination-item-link"
          disabled={true}
        >
          <span
            aria-label="left"
            className="anticon anticon-left"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="left"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
              />
            </svg>
          </span>
        </a>
      </li>
      <li
        className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
        title={1}
      >
        <a>
          1
        </a>
      </li>
      <li
        aria-disabled={true}
        className="ant-pagination-next ant-pagination-disabled"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={null}
        title="Next Page"
      >
        <a
          className="ant-pagination-item-link"
          disabled={true}
        >
          <span
            aria-label="right"
            className="anticon anticon-right"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
              />
            </svg>
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Products List component snapshot testing should render the component 1`] = `
<div
  className="ant-list ant-list-split ant-list-something-after-last-item"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-list-empty-text"
      >
        <div
          className="ant-empty ant-empty-normal"
        >
          <div
            className="ant-empty-image"
          >
            <svg
              className="ant-empty-img-simple"
              height="41"
              viewBox="0 0 64 41"
              width="64"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fillRule="evenodd"
                transform="translate(0 1)"
              >
                <ellipse
                  className="ant-empty-img-simple-ellipse"
                  cx="32"
                  cy="33"
                  rx="32"
                  ry="7"
                />
                <g
                  className="ant-empty-img-simple-g"
                  fillRule="nonzero"
                >
                  <path
                    d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                  />
                  <path
                    className="ant-empty-img-simple-path"
                    d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                  />
                </g>
              </g>
            </svg>
          </div>
          <p
            className="ant-empty-description"
          >
            No Data
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-list-pagination"
  >
    <ul
      className="ant-pagination"
      style={Object {}}
      unselectable="unselectable"
    >
      <li
        aria-disabled={true}
        className="ant-pagination-prev ant-pagination-disabled"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={null}
        title="Previous Page"
      >
        <a
          className="ant-pagination-item-link"
          disabled={true}
        >
          <span
            aria-label="left"
            className="anticon anticon-left"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="left"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
              />
            </svg>
          </span>
        </a>
      </li>
      <li
        className="ant-pagination-item ant-pagination-item-0 ant-pagination-disabled ant-pagination-item-disabled"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
        title={0}
      >
        <a>
          0
        </a>
      </li>
      <li
        aria-disabled={true}
        className="ant-pagination-next ant-pagination-disabled"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={null}
        title="Next Page"
      >
        <a
          className="ant-pagination-item-link"
          disabled={true}
        >
          <span
            aria-label="right"
            className="anticon anticon-right"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
              />
            </svg>
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
